unison-cli/src/Unison/CommandLine/Globbing.hs

```diff
@@ -121,10 +118,6 @@ expandGlobs targets rootBranch currentPath s = either recover Just $ do
         | isAbsolute = (Path.Absolute . Path.unrelative) <$> paths
         | otherwise = Path.resolve currentPath <$> paths
   pure (Path.convert <$> relocatedPaths)
-  where
-    recover = \case
```

Comment:
I'm not sure why we were returning zero matches rather than failing completely on `NoTargets` before. I can't think of any reason to do that.